### PR TITLE
Tests for certificates with SAN

### PR DESCRIPTION
**Pull request by apophys**
Commits include several new test cases for CA ACLs and cert request for CSRs containing subject alternative name extension.

Also included minor fixes in used tracker and couple of new context managers used in the test cases.

### CODE REVIEW

File: ipatests/test_xmlrpc/test_caacl_profile_enforcement.py [hunk 1]
```diff
@@ -354,3 +362,234 @@ def test_sign_smime_csr_fallback_to_default_cert_profile(
...
+
+    The class prepares an environment for test cases based
+    on evaluation of ACLs and fields requested in a CSR.
+
+    The class creates following entries:
+
+        * two host entries
+            * santest-host-1
+            * santest-host-2
+        * one service entry
```
Comment (martbab) [inline ipatests/test_xmlrpc/test_caacl_profile_enforcement.py]: Aren't there two service entries? (santest_service_host_1 and santest_service_host_2)

File: ipatests/test_xmlrpc/test_caacl_profile_enforcement.py [hunk 2]
```diff
@@ -354,3 +362,234 @@ def test_sign_smime_csr_fallback_to_default_cert_profile(
...
+
+
+@pytest.mark.tier1
+class TestSignServiceCertWithoudSANServiceInACL(CAACLEnforcementOnCertBase):
+    """ Sign certificate request with multiple subject alternative names
+
+    This test case doesn't have the service hosted on a host in SAN
+    in the CA ACL. The assumption is that the issuance will fail.
+
+    Using the environment of the base class, modify the service to be managed
```
Comment (martbab) [inline ipatests/test_xmlrpc/test_caacl_profile_enforcement.py]: Copy-paste error here?

File: ipatests/util.py [hunk 3]
```diff
@@ -720,6 +726,42 @@ def change_principal(user, password, client=None, path=None,
...
+    The keytab will generate new keys if not specified
+    otherwise in the options.
+    To retrieve existing keytab, use the -r option
+    """
+    keytab_filename = os.path.join('/tmp', str(uuid.uuid4()))
+
+    try:
+        cmd = [paths.IPA_GETKEYTAB, '-p', principal, '-k', keytab_filename]
+
+        if options:
```
Comment (apophys) [inline ipatests/util.py]: keytab_filename is a file that will be created by ipa-getkeytab command, what is there to check?

File: ipatests/util.py [hunk 4]
```diff
@@ -709,8 +711,12 @@ def change_principal(user, password, client=None, path=None,
 
     try:
         with private_ccache(ccache_name):
-            kinit_password(user, password, ccache_name,
-                           canonicalize=canonicalize, enterprise=enterprise)
+            if keytab:
```
Comment (apophys) [inline ipatests/util.py]: Is here a typo or word missing? I do not understand. If one specifies keytab I would assume the tester knows what he's doing.

I can document the context manager though.

**Comment by martbab** [comment]
NACK: you probably forgot to add service fixtures as params to the added test cases: https://paste.fedoraproject.org/437721/51355181/

In addition please write sensible commit message to commit f43833d and probably squash the last commit into 2d75883

I have also noticed that you linked the commits to a ticket in a already closed milestone. Per our process guidelines you need to open a new ticket and go through a new triage, sorry.

**Comment by apophys** [comment]
I have fixed typos and implemented the proposed test cases. I have also provided docstring to the change_principal context manager.

**Comment by martbab** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/4f8e212c428b1fff63f38ea15d7ce9f230085c5c
https://fedorahosted.org/freeipa/changeset/7eb78aa8dbc167c36869ddd6cfa525139aff7bbc
https://fedorahosted.org/freeipa/changeset/10b4b155b6b411ab339bce92c1a335b4914cfc1c
ipa-4-4:
https://fedorahosted.org/freeipa/changeset/e607bd000b1593f6824ccc9ca8862eec43c442bb
https://fedorahosted.org/freeipa/changeset/3fd233458bc7976790d48374c48d48f27596536e
https://fedorahosted.org/freeipa/changeset/5d75842017a917ac7769e5c837bfffa1ba0e1c74

**Pull request closed without merging**